app/controllers/api/v2/app_groups_controller.rb

```diff
@@ -166,25 +166,48 @@ def update_cost
       affected_app: affected_app
     }, status: :ok and return
   end
+
   def deactivated_by_cluster_name
-    @helm_infrastructure = HelmInfrastructure.find_by(cluster_name: params[:cluster_name])
-    
+    cluster_name = params[:cluster_name]
+    app_group_name = params[:name]
+  
+    # Validate presence of both cluster_name and app_group_name
+    unless cluster_name.present? && app_group_name.present?
+      render(json: {
+        success: false,
+        errors: ['Both cluster_name and app_group_name are required'],
+        code: 400,
+      }, status: :bad_request) && return
+    end
+  
+    @helm_infrastructure = HelmInfrastructure.find_by(cluster_name: cluster_name)
```

Comment:
btw, should directly match both `cluster_name` and ` app_group_name` here and remove this line of codes 
```    # Check if the Helm Infrastructure's app_group_name matches the provided app_group_name
    unless @helm_infrastructure.app_group.name == app_group_name
      render(json: {
        success: false,
        errors: ['Mismatched app_group_name and cluster_name'],
        code: 400,
      }, status: :bad_request) && return
    end```
    
    or should keep as it is?
    wdyt guys @ce314007 @clavinjune

Reply:
i cant find `app_group_name` in helm Infrastructure db schema? how to call it both? bang @fadlinurhasan

Reply:
we can do that as it is or directly match both fields using join.

but I guess @ce314007 alr use join here, we can go on with that. but need to be aware that using join is not more effective